dotnet-estimation/dotnet-estimation/Templates/WebAPI/Devon4Net.Application.WebAPI.Implementation/Business/SessionManagement/Controllers/SessionController.cs

```diff
@@ -0,0 +1,61 @@
+using System.ComponentModel.DataAnnotations;
+using Newtonsoft.Json;
+using Devon4Net.Application.WebAPI.Implementation.Business.SessionManagement.Service;
+using Devon4Net.Application.WebAPI.Implementation.Business.SessionManagement.Dtos;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Authorization;
+using Devon4Net.Application.WebAPI.Implementation.Domain.Entities;
+using Devon4Net.Application.WebAPI.Implementation.Business.SessionManagement.Converters;
+
+using Devon4Net.Infrastructure.Logger.Logging;
+using Devon4Net.Application.WebAPI.Implementation.Business.SessionManagement.Exceptions;
+
+namespace Devon4Net.Application.WebAPI.Implementation.Business.SessionManagement.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    [EnableCors("CorsPolicy")]
+    public class SessionController : ControllerBase
+    {
+        private readonly ISessionService _sessionService;
+
+        public SessionController(ISessionService SessionService)
+        {
+            _sessionService = SessionService;
+        }
+
+
+        /// <summary>
+        /// Remove a Session user
+        /// </summary>
+        /// <returns></returns>
+        [HttpPut]
+        [AllowAnonymous]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [Route("/estimation/v1/session/{sessionId:long}/leaveSession/{userId}")]
+        public async Task<IActionResult> RemoveUserFromSession(long sessionId, String userId)
+        {
+            Devon4NetLogger.Debug($"Put-Request for removing user with id: {userId} from session status with id: {sessionId}");
+
+            try
+            {
+                var leaveResult = await _sessionService.RemoveUserFromSession(sessionId, userId);
+
+                return new ObjectResult(JsonConvert.SerializeObject(leaveResult));
+            }
+            catch (Exception exception)
+            {
+                return exception switch
+                {
+                    NotFoundException _ => NotFound(),
```

Comment:
As far as I know you have to throw the `NotFoundException` explicitly because it's a exception class that we've created ourselves.
So you should either remove this line here (and the `ProducesResponseType` attribute for Status404NotFound in line 38) or throw the exception somewhere in the SessionService.
Please correct me if I'm wrong.